employee-rostering-frontend/src/store/alert/operations.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from 'react';
+import { Text, Title, Button, ButtonVariant, Modal } from '@patternfly/react-core';
+import { ServerSideExceptionInfo } from 'types';
+import * as actions from './actions';
+import { AlertInfo, AddAlertAction, RemoveAlertAction } from './types';
+import { useTranslation } from 'react-i18next';
+
+interface ObjectStringMap { [K: string]: any }
+
+export function showInfoMessage(i18nKey: string, params?: ObjectStringMap, components?: JSX.Element[]): AddAlertAction {
+  return showMessage("info", i18nKey, params, components);
+}
+
+export function showSuccessMessage(i18nKey: string, params?: ObjectStringMap, components?: JSX.Element[]): AddAlertAction {
+  return showMessage("success", i18nKey, params, components);
+}
+
+export const ServerSideExceptionDialog: React.FC<React.PropsWithChildren<ServerSideExceptionInfo>> = (props) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const dialogBody = createStackTrace(props);
+  const { t } = useTranslation();
+
+  return (
+    <>
+      {t(props.i18nKey, props.messageParameters)}
+      <Button
+        aria-label="Show Stack Trace"
+        variant={ButtonVariant.link}
+        onClick={() => setIsOpen(true)}
+      >
+        {props.children}
+      </Button>
+      <Modal
+        title="Server Side Error"
+        isOpen={isOpen}
+        onClose={() => setIsOpen(false)}
+        actions={(
+          <Button
+            aria-label="Close"
+            onClick={() => setIsOpen(false)}
+          >
+            Close
+          </Button>
+        )}
+      >
+        {dialogBody}
+      </Modal>
+    </>
+  );
+};
+
+export function showServerError(exceptionInfo: ServerSideExceptionInfo): AddAlertAction {
+  return addAlert({
+    i18nKey: "exception",
+    variant: "danger",
+    params: {
+      message: exceptionInfo.exceptionMessage,
+    },
+    components: [<ServerSideExceptionDialog {...exceptionInfo} key="0" />]
```

Comment:
Here. Why not to use just the info as the action payload?

Reply:
See above: we need to tell `Alerts` to render a `Link` instead of text.